torch/_dynamo/variables/builder.py

```diff
@@ -1166,18 +1166,29 @@ def wrap_to_fake_tensor_and_record(
         if tx.fake_mode.shape_env is not None:
             dynamic_dims = []
             constraint_dims = []
+            automatic_marked_dims = set()
+            if isinstance(source, LocalSource):
```

Comment:
Suppose that I pass in a tensor in a tuple to a function. Then the source will not be a LocalSource; it will be a tuple access on a LocalSource. Don't I still want to be able to autodetect dynamic in htis case?